src/background.c

```diff
@@ -262,6 +262,83 @@ handle_notify_background (XdpImplBackground *object,
   return TRUE;
 }
 
+static gboolean
+handle_enable_autostart (XdpImplBackground *object,
+                         GDBusMethodInvocation *invocation,
+                         const char *arg_app_id,
+                         gboolean    arg_enable,
+                         const char * const *arg_commandline,
+                         gboolean    arg_activatable)
+{
+  gboolean result = FALSE;
+  g_autofree char *dir = NULL;
+  g_autofree char *file = NULL;
+  g_autofree char *path = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autofree char *commandline = NULL;
+  g_autoptr(GKeyFile) keyfile = NULL;
+
+  g_debug ("background: handle EnableAutostart");
+
+  file = g_strconcat (arg_app_id, ".desktop", NULL);
+  dir = g_build_filename (g_get_user_config_dir (), "autostart", NULL);
+  path = g_build_filename (dir, file, NULL);
+
+  if (!arg_enable)
+    {
+      unlink (path);
+      g_debug ("Removed %s", path);
+      goto out;
+    }
+
+  if (g_mkdir_with_parents (dir, 0755) != 0)
+    {
+      g_warning ("Failed to create dirs %s", dir);
+      goto out;
+    }
+
+  commandline = g_strjoinv (" ", (char **)arg_commandline);
```

Comment:
I think this is where we should do any required escaping, rather than in the portal.